### PR TITLE
Closes #1458 derive param exist flag@devel

### LINKED ISSUE #1458: General Issue: Increase test coverage for `derive_param_exist_flag.R`
### Background Information

We need to increase test coverage for our functions. The below screenshot has the function and gap in test coverage that we need to write additional unit tests for derive_param_exist_flag.

image

### Definition of Done

New tests written that covers the red gap area for derive_param_exist_flag.R.

Use devtools::test_coverage() to reproduce the report

### PR DESCRIPTION
**Pull request by jeffreyad**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by bms63** [comment]
Hey Jeff.  You are need to change the branch to merge into - `devel` and not `main` .  I always look at the total number of files changed for a sanity check :)  
image

File: tests/testthat/test-derive_param_exist_flag.R [hunk 1]
```diff
@@ -84,3 +84,26 @@ test_that("derive_param_exist_flag Test 2: error is issued if aval_fun returns w
...
+      dataset = adrs,
+      dataset_adsl = adsl,
+      dataset_add = adrs,
+      filter_add = PARAMCD == "OVR",
+      condition = AVALC == "PD",
+      false_value = "N",
+      set_values_to = vars(
+        PARAMCD = "OVR",
+        ANL01FL = "Y"
+      )
```
Comment (thomas-neitmann) [inline tests/testthat/test-derive_param_exist_flag.R:102]: No need to assign the output to a variable here.

```suggestion
    derive_param_exist_flag(
      dataset = adrs,
      dataset_adsl = adsl,
      dataset_add = adrs,
      filter_add = PARAMCD == "OVR",
      condition = AVALC == "PD",
      false_value = "N",
      set_values_to = vars(
        PARAMCD = "OVR",
        ANL01FL = "Y"
      )
```

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]
Thanks a lot @jeffreyad! Just a minor comment from my side.

**Comment by jeffreyad** [comment]
> Thanks a lot @jeffreyad! Just a minor comment from my side.

Thanks, updated.

File: renv.lock [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "R": {
-    "Version": "3.6.3",
+    "Version": "4.0.3",
```
Comment (thomas-neitmann) [inline renv.lock:3]:
```suggestion
    "Version": "3.6.3",
```

**Review by thomas-neitmann — APPROVED** [review]
Thanks a lot @jeffreyad, much appreciated!

**Pull request merged**